tools/ctl/ctl.c

```diff
@@ -571,12 +695,21 @@ int main(int argc, char *argv[])
 
 	/* Just print the ABI header if requested */
 	if (ctl_data->print_abi_header) {
-		ctl_data->ctrl_size = sizeof(struct sof_abi_hdr);
+		if (ctl_data->ipc_version == 4)
+			ctl_data->ctrl_size = sizeof(struct sof_ipc4_abi_hdr);
+		else /* only ipc3 and ipc4 is supported */
+			ctl_data->ctrl_size = sizeof(struct sof_abi_hdr);
 		buffer_alloc(ctl_data);
 		header_init(ctl_data);
-		hdr = (struct sof_abi_hdr *)
-			&ctl_data->buffer[BUFFER_ABI_OFFSET];
-		hdr->size = ctl_data->print_abi_size;
+		if (ctl_data->ipc_version == 4) {
```

Comment:
can we use either a switch of ops callback. That will prevent partial implementations if for some reason we move to another version in the future